extends Level

func _ready() -> void:
	var mansplaining_recovery := Choose2.new(
		"Agni: Why would you say that?! The world is already so messed up it doesnâ€™t even matter.  Worry about yourself.",
		DialogNode.new(
			"$name: Hey, Iâ€™m sorry",
			DialogNode.new(
				"$name: I just wanted to show you that there are [i]some[/i] things we can do, even if it feels meaningless in the big picture, it can improve our local community.",
				DialogNode.new(
					"Agni: Ugh, thatâ€™s fair, I guess.  Iâ€™ll be honest, I still get kind of confused about what things go in which cans, itâ€™s too complicated.",
					DialogNode.new(
						"$name: Itâ€™s all good! How about we clean up our picnic together?",
						DialogNode.new(
							"Agni: Hey, thatâ€™d be nice.",
							null,
							null,
							add_heart,
						)
					)
				)
			)
		),
		"I'm sorry",
		DialogNode.new(
			"$name: But we still have to try.  Iâ€™m going to sort my trash here â€“ it doesnâ€™t cost me anything â€“ and Iâ€™d love it if you joined me.",
			DialogNode.new(
				"Agni: Why not, I guessâ€¦",
				Terminus.new()
			)
		),
		"You still have to try",
		null,
		func(player): player.hearts -= 1; explode("ðŸ’”"),
	)
	
	var recycling := Choose2.new(
		"*** Agni finishes their water bottle, and throws it away in the trash bin ***",
		DialogNode.new(
			"$name: You canâ€™t really say that when you donâ€™t even recycle properly.",
			mansplaining_recovery,
		),
		"You could at least recycle.",
		DialogNode.new(
			"It doesnâ€™t cost us anything to do small things like recycling.  Iâ€™m going to sort the trash from our picnic, and Iâ€™d love it if you joined me!",
			DialogNode.new(
				"Agni: I guess I see your point. I canâ€™t say Iâ€™ve been on a trash-sorting date beforeâ€¦",
				Terminus.new()
			)
		),
		"Doing small things still helps"
	)
	
	var weather := DialogNode.new(
		"$name: Some crazy weather weâ€™ve been having, huh? We didn't have all these floods and wildfires and stuff a few years agoâ€¦",
		DialogNode.new(
			"Agni: Ugh, itâ€™s really damn annoying talking about this.  Thereâ€™s nothing we can do - people messed up the world and we have to suffer the consequences.",
			recycling
		)
	)
	
	var food_offer_options := Choose3.new(
		"Agni: Hey, your rice bowl doesnâ€™t look bad either.",
		DialogNode.new(
			"$name: Would you like some of mine?",
			DialogNode.new(
				"Agni: Only if you try some of mine.  When people baby me, I canâ€™t stand it.",
				DialogNode.new(
					"$name: Sounds good!",
					weather
				)
			)
		),
		"Offer a bite",
		DialogNode.new(
			"$name: Would you like to split?",
			DialogNode.new(
				"Agni:  Hey, sure.",
				DialogNode.new(
					"Agni: (+1 HEART) Cute of you to keep the relationship equal.",
					weather,
					null,
					add_heart
				)
			)
		),
		"Offer to split",
		weather,
		"Do nothing",
	)
	
	tree = DialogNode.new(
		"After texting back and forth, you organized a picnic with Agni.",
		DialogNode.new(
			"Agni: Hey $name.",
			DialogNode.new(
				"$name: Hi Agni! Iâ€™ve been looking forward to this!",
				DialogNode.new(
					"You spread out the picnic blanket and begin eating.",
					DialogNode.new(
						"$name: So what is it that youâ€™re eating?",
						DialogNode.new(
							"Agni: I picked up this char-grilled chicken caesar salad from a cafÃ© by my house.",
							DialogNode.new(
								"$name: It looks so good.",
								food_offer_options
							)
						)
					),
					null,
					func(_p): $TextureRect.texture = preload("res://TRASH_GRASS_PICNIC.png"),
				)
			),
			AGNI,
		)
	)
	super._ready()
